src/tags/control/DateTime.js

```diff
@@ -198,6 +203,15 @@ const Model = types
       self.time = undefined;
     },
 
+    validDateFormat(dateString) {
+      const dateNumberArray = dateString.split("-").map(dateString => parseInt(dateString, 10));
+      const year = dateNumberArray[0];
+      const jsDate = new Date(dateString);
+
+      if (isNaN(jsDate) || year < 1000 || year >= 10000) return false;
```

Comment:
What range of date years are supported for this? Can I as a user enter a date in the past or future? If so, why the valid year range of 1001-9999?

Reply:
there are valid dates that are not valid to our system the year must be four positive integers and this is to check that, I will make more readable.